api/nodes/serializers.py

```diff
@@ -15,15 +15,18 @@ class NodeSerializer(JSONAPISerializer):
     date_modified = ser.DateTimeField(read_only=True)
 
     links = LinksField({
```

Comment:
I'm not sure. LinkedSerializer seems like less boilerplate for additional Link types. For example, I'm going to have to get links to Waterbutler resources, which is going to be a somewhat different process to the way you're doing other links. To add Waterbutler links, I've added a new class which is somewhat filled in but mostly not:

```py
class WaterbutlerLink(object):
    """Link object to use in conjunction with Links field. Builds a Waterbutler URL for files.
    """

    def __init__(self, endpoint, args=None, kwargs=None, **kw):
        # self.endpoint = endpoint
        self.kwargs = kwargs or {}
        self.args = args or tuple()
        self.reverse_kwargs = kw

    # TODO Build this or use the function from the Folder PR
    def resolve_url(self, obj):
        return "TBD"
```

And I had to add the elif to determine if it's a WaterbutlerLink. I think just having the get_links in the NodeFileSerializer may make things a bit simpler and more readable, but this should not be considered strong analysis by any stretch.